### PR TITLE
LibJS: Support all line terminators (LF, CR, LS, PS)

### PR DESCRIPTION
**Pull request by linusg**
You can now feed weirdo line terminators to LibJS. Fixes 7 test262-parser-tests tests.

https://tc39.es/ecma262/#sec-line-terminators

### CODE REVIEW

File: Libraries/LibJS/Lexer.cpp [hunk 1]
```diff
@@ -30,8 +30,27 @@
...
+    } while (0)
+
+#define THREE_CHARS_VIEW()                                                       \
+    StringView view;                                                             \
+    do {                                                                         \
+        char second_char = m_source[m_position];                                 \
+        char third_char = m_source[m_position + 1];                              \
+        char three_chars[] { (char)m_current_char, second_char, third_char, 0 }; \
+        view = { three_chars };                                                  \
+    } while (0)
```
Comment (linusg) [inline Libraries/LibJS/Lexer.cpp:52]: This is what I asked about on IRC this morning - made it two macros instead, I guess that works?!

File: Libraries/LibJS/Tests/parser-line-terminators.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    }).toThrowWithMessage(SyntaxError, "line: 3, column: 1");
+});
+
+test("PARAGRAPH SEPARATOR is a line terminator", () => {
+    expect(() => {
+        Function("\
\
@");
+    }).toThrowWithMessage(SyntaxError, "line: 3, column: 1");
+});
```
Comment (linusg) [inline Libraries/LibJS/Tests/parser-line-terminators.js:29]: GitHub doesn't show it but there are LS/PS behind each backslash.

File: Libraries/LibJS/Tests/parser-line-terminators.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+test("PARAGRAPH SEPARATOR is a line terminator", () => {
+    expect(() => {
+        Function(`

@`);
+    }).toThrowWithMessage(SyntaxError, "line: 3, column: 1");
+});
```
Comment (linusg) [inline Libraries/LibJS/Tests/parser-line-terminators.js:29]: GitHub doesn't show it, but the template literal contains two LS/PS chars.

File: Libraries/LibJS/Lexer.cpp (hunk 1 above)
Comment (alimpfard) [reply Libraries/LibJS/Lexer.cpp:52]: Any particular reason why you can't just use `m_source`?
`StringView view { &m_source[m_position-1], 2 }` for the two chars one?
(What even is `m_source`? a String?)

Comment (linusg) [reply Libraries/LibJS/Lexer.cpp:52]: > Any particular reason why you can't just use m_source?

'cause I didn't think of that, that's why :smile:

> (What even is m_source? a String?)

A StringView :)

I'll try this - thanks!

Comment (linusg) [reply Libraries/LibJS/Lexer.cpp]: that totally worked lol. To not see the forest for the trees, or something... ty!

**Comment by linusg** [comment]
A little JavaScript speciality I initially missed but fixed just now: LF/CR in string literal are not ok, LS/PS *are* :^)

**Pull request merged**